packages/sats-wagmi/src/connectors/unisat.ts

```diff
@@ -200,11 +200,16 @@ class UnisatConnector extends SatsConnector {
         inputs.push(index);
       }
     }
+
+    if (!this.paymentAddress) {
+      throw new Error('No payment address specified');
+    }
+
     const toSignInputs = inputs.map((index) => {
       return {
         index,
         publicKey: this.publicKey,
-        disableTweakSigner: true
+        disableTweakSigner: this.getAddressType(this.paymentAddress!) !== AddressType.p2tr
```

Comment:
without it signer throws that public key is not allowed to sign output -> for taproot addresses we need to pass `false` -> enable tweak signer

Reply:
More similar changes will be added to other contectors

Reply:
I think this can probably be hard coded to false, right? Or just omitted, since the default is false as well. Or are there cases where we want to need the untweaked taproot key?

Also, do we need to use the  `tweakedPublicKey` rather than `publicKey`? https://github.com/leather-io/extension/issues/4125#issuecomment-1929391022

Reply:
> I think this can probably be hard coded to false, right? Or just omitted, since the default is false as well. 
Made it like this in order to not break functionality that might depend on `disableTweakSigner: true` as it was set previously
> Or are there cases where we need the untweaked taproot key?
not sure about that
> Also, do we need to use the tweakedPublicKey rather than publicKey?
I do not see a property which we can stick unweaked taproot key into. will investigate

Reply:
In this comment user report the exact issue I was receiving when using public key (not tweaked) with `disableTweakSigner` set to `false`. [link](https://github.com/unisat-wallet/extension/issues/86#issuecomment-1705698196)

And according to that [reply](https://github.com/unisat-wallet/extension/issues/86#issuecomment-1752020625) tweaked public key is used by default when signing and unlocking Taproot addresses

Reply:
I'm not sure I agree with you. Will we be able to test this on a private branch before it gets rolled out to the public?